frontend/app_flowy/lib/workspace/presentation/settings/widgets/settings_file_system_view.dart

```diff
@@ -0,0 +1,167 @@
+}
+
+@visibleForTesting
+class SettingsFileSystemViewState extends State<SettingsFileSystemView> {
+  final _locationCubit = SettingsLocationCubit()..fetchLocation();
+
+  @override
+  Widget build(BuildContext context) {
+    return ListView.separated(
+      itemBuilder: (context, index) {
+        if (index == 0) {
+          return _buildLocationCustomizer();
+        } else if (index == 1) {
+          // return _buildExportDatabaseButton();
+        }
+        return Container();
+      },
+      separatorBuilder: (context, index) => const Divider(),
+      itemCount: 2, // make the divider taking effect.
+    );
+  }
+
+  // Customize Default Location
+  Widget _buildLocationCustomizer() {
+    return BlocProvider<SettingsLocationCubit>.value(
+      value: _locationCubit,
+      child: BlocBuilder<SettingsLocationCubit, SettingsLocation>(
+          builder: (context, state) {
+        return ListTile(
+          title: FlowyText.regular(
+            LocaleKeys.settings_files_defaultLocation.tr(),
+            fontSize: 15.0,
+          ),
+          subtitle: Tooltip(
+            message: LocaleKeys.settings_files_doubleTapToCopy.tr(),
+            child: GestureDetector(
+              onDoubleTap: () {
+                Clipboard.setData(ClipboardData(
+                  text: state.path,
+                ));
+              },
+              child: FlowyText.regular(
+                state.path ?? '',
+                fontSize: 10.0,
+              ),
+            ),
+          ),
+          trailing: Row(
+            mainAxisSize: MainAxisSize.min,
+            children: [
+              Tooltip(
+                message: LocaleKeys.settings_files_restoreLocation.tr(),
+                child: FlowyIconButton(
+                  icon: const Icon(Icons.restore_outlined),
+                  onPressed: () async {
+                    final result = await appFlowyDocumentDirectory();
+                    await _setCustomLocation(result.path);
+                    await FlowyRunner.run(
+                      FlowyApp(),
+                      config: const LaunchConfiguration(
+                        autoRegistrationSupported: true,
+                      ),
+                    );
+                    if (mounted) {
+                      Navigator.of(context).pop();
+                    }
+                  },
+                ),
+              ),
+              const SizedBox(
+                width: 5,
+              ),
+              Tooltip(
+                message: LocaleKeys.settings_files_customizeLocation.tr(),
+                child: FlowyIconButton(
+                  icon: const Icon(Icons.folder_open_outlined),
+                  onPressed: () async {
+                    final result = await FilePicker.platform.getDirectoryPath();
+                    if (result != null) {
+                      await _setCustomLocation(result);
+                      await reloadApp();
+                    }
```

Comment:
Would you mind refactoring this bracket. It 's not easy to find out where am I.

Reply:
separated this part into a new file as a new `widget`.